src/mon/OSDMonitor.cc

```diff
@@ -10194,6 +10194,7 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
     } else {
       //NOTE:for replicated pool,cmd_map will put rule_name to erasure_code_profile field
       rule_name = erasure_code_profile;
+      erasure_code_profile = "";
```

Comment:
could you update `OSDMonitor::prepare_new_pool()` also ? so we don't update `pi->erasure_code_profile` with the given `erasure_code_profile` if `pool_type == pg_pool_t::TYPE_REPLICATED`?